rest/rest-api/src/main/java/org/apache/brooklyn/rest/domain/EntityConfigSummary.java

```diff
@@ -18,43 +18,82 @@
-
-import org.apache.brooklyn.config.ConfigKey;
-
 import java.net.URI;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
+import javax.annotation.Nullable;
+
+import org.apache.brooklyn.config.ConfigKey;
+import org.apache.brooklyn.util.text.StringPredicates;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+
 public class EntityConfigSummary extends ConfigSummary {
 
     private static final long serialVersionUID = -1336134336883426030L;
 
     @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
+    private final Boolean pinned;
+
+    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
+    private final List<String> constraints;
+
+    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
     private final Map<String, URI> links;
 
     public EntityConfigSummary(
             @JsonProperty("name") String name,
             @JsonProperty("type") String type,
             @JsonProperty("description") String description,
             @JsonProperty("defaultValue") Object defaultValue,
-            @JsonProperty("reconfigurable") boolean reconfigurable,
+            @JsonProperty("reconfigurable") Boolean reconfigurable,
             @JsonProperty("label") String label,
             @JsonProperty("priority") Double priority,
             @JsonProperty("possibleValues") List<Map<String, String>> possibleValues,
+            @JsonProperty("pinned") Boolean pinned,
+            @JsonProperty("constraints") List<String> constraints,
             @JsonProperty("links") Map<String, URI> links) {
         super(name, type, description, defaultValue, reconfigurable, label, priority, possibleValues);
+        this.pinned = pinned;
+        this.constraints = (constraints == null) ? ImmutableList.<String>of() : ImmutableList.copyOf(constraints);
         this.links = (links == null) ? ImmutableMap.<String, URI>of() : ImmutableMap.copyOf(links);
     }
 
-    public EntityConfigSummary(ConfigKey<?> config, String label, Double priority, Map<String, URI> links) {
+    public EntityConfigSummary(ConfigKey<?> config, String label, Double priority, Boolean pinned, Map<String, URI> links) {
         super(config, label, priority);
+        this.pinned = pinned;
+        this.constraints = ImmutableList.copyOf(Iterables.transform(config.getConstraints(), new Function<Predicate, String>() {
+            @Nullable
+            @Override
+            public String apply(@Nullable Predicate predicate) {
+                if (predicate == null) {
+                    return null;
+                } else if (predicate.getClass().getName().equals(StringPredicates.isNonBlank().getClass().getName())) {
```

Comment:
Minor,  could assign `StringPredicates.isNonBlank().getClass().getName()`  to a static field of the anonymous class.